src/Metadata/Property/PropertyMetadata.php

```diff
@@ -281,14 +255,32 @@ public function withAttributes(array $attributes): self
         return $metadata;
     }
 
+    /**
+     * Gets child inherited.
+     */
+    public function getChildInherited(): ?string
+    {
+        return $this->childInherited;
+    }
+
+    /**
+     * Is the property inherited from a child class?
+     */
+    public function hasChildInherited(): bool
+    {
+        return null !== $this->childInherited;
+    }
+
     /**
      * Is the property inherited from a child class?
      *
-     * @return string|null
+     * @deprecated since version 2.4, to be removed in 3.0.
      */
-    public function isChildInherited()
+    public function isChildInherited(): ?string
     {
-        return $this->childInherited;
+        @trigger_error(sprintf('The use of "%1$s::isChildInherited()" is deprecated since 2.4 and will be removed in 3.0. Use "%1$s::getChildInherited()" or "%1$s::hasChildInherited()" directly instead.', __CLASS__), E_USER_DEPRECATED);
+
+        return $this->getChildInherited();
```

Comment:
why the change?

Reply:
The name of the method and the return type are inconsistent... It's a nonsense.